packages/devtools_app/lib/src/timeline/flutter/timeline_flame_chart.dart

```diff
@@ -1,17 +1,349 @@
 // Copyright 2019 The Chromium Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
+import 'dart:async';
+import 'dart:math' as math;
 
 import 'package:flutter/material.dart';
 
-class TimelineFlameChart extends StatelessWidget {
+import '../../charts/flutter/flame_chart.dart';
+import '../../flutter/controllers.dart';
+import '../../ui/colors.dart';
+import '../../ui/theme.dart';
+import '../../utils.dart';
+import '../timeline_controller.dart';
+import '../timeline_model.dart';
+
+class TimelineFlameChart extends StatefulWidget {
+  @override
+  _TimelineFlameChartState createState() => _TimelineFlameChartState();
+}
+
+class _TimelineFlameChartState extends State<TimelineFlameChart> {
+  TimelineController _controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    _controller = Controllers.of(context).timeline;
```

Comment:
If you're not subscribing to the controller here, you can drop these references to it.

Reply:
done. made TimelineFlameChart a stateless widget